src/aws/osml/model_runner/inference/http_detector.py

```diff
@@ -21,16 +23,80 @@
 logger = logging.getLogger(__name__)
 
 
+class CountingRetry(urllib3.Retry):
+    def __init__(self, *args, **kwargs):
+        """
+        Retry class implementation that counts the number of retries.
+
+        :return: None
+        """
+        super(CountingRetry, self).__init__(*args, **kwargs)
+        self.retry_counts = 0
+
+    def increment(self, *args, **kwargs) -> Retry:
```

Comment:
I don't think this increment function works as intended. Looking at the urllib3.Retry class their increment method updates the history and all the counts on the returned object. This method increments the result on the current object but the new retry that is returned will always have a retry_count of 0. 

Line 41 should be something like: `result.retry_counts = self.retry_counts +1`

It's also not clear to me that we actually need to implement this class. Can't we just use the length of the history on the urllib3.Retry class to find out how many retries occurred?

Reply:
I initially had an implementation using the history and I realized that if this fails on on of the retries you never get a response with a history. I found online recommendations to track it in the Retry object directly but I could revert to that approach.